src/Surfnet/StepupGateway/GatewayBundle/Entity/SamlEntity.php

```diff
@@ -110,6 +109,16 @@ public function toServiceProvider()
         $configuration['entityId']             = $this->entityId;
         $configuration['configuredLoas']       = $decodedConfiguration['loa'];
 
+        $configuration['secondFactorOnly'] = false;
+        if (isset($decodedConfiguration['second_factor_only'])) {
+            $configuration['secondFactorOnly'] = (bool) $decodedConfiguration['second_factor_only'];
```

Comment:
Right, but if you don't provide a default then you have an undefined index when it is checked in the ```ServiceProvider``` class.